dbms/src/Flash/Planner/Plans/PhysicalAggregationBuild.cpp

```diff
@@ -77,6 +84,39 @@ void PhysicalAggregationBuild::buildPipelineExecGroupImpl(
 EventPtr PhysicalAggregationBuild::doSinkComplete(PipelineExecutorContext & exec_context)
 {
     assert(aggregate_context);
+
+    SCOPE_EXIT({ aggregate_context.reset(); });
+    if (!aggregate_context->hasSpilledData())
```

Comment:
This check should at least moved to after `aggregate_context->getAggSpillContext()->finishSpillableStage();` How about move the whole code block after check `AggregateFinalSpillEvent`?

Reply:
Done.